src/Components/Aspire.Azure.Messaging.EventHubs/README.md

```diff
@@ -0,0 +1,182 @@
+
+## Getting started
+
+### Prerequisites
+
+- Azure subscription - [create one for free](https://azure.microsoft.com/free/)
+- Azure Event Hubs namespace, learn more about how to [add an Event Hubs namespace](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create). Alternatively, you can use a connection string, which is not recommended in production environments.
+
+### Install the package
+
+Install the .NET Aspire Azure Event Hubs library with [NuGet](https://www.nuget.org):
+
+```dotnetcli
+dotnet add package Aspire.Azure.Messaging.EventHubs
+```
+
+## Supported clients with Options classes
+
+The following clients are supported by the library, along with their corresponding Options classes:
+
+| Client Type [index]              | Options Class                 |
+|------------------------------------------|-------------------------------|
+| EventHubProducerClient[1] | EventHubProducerClientOptions |
+| EventHubConsumerClient[2] | EventHubConsumerClientOptions |
+| EventProcessorClient[3]   | EventProcessorClientOptions   |
+| PartitionReceiver[4]      | PartitionReceiverOptions      |
+
+### AzureMessagingEventHubsSettings shared configuration
+
+The shared settings for all clients are suffixed below with one or more
+indices corresponding to the client types listed above that may use them:
+
+| Member                                                                                   | Type             |
+|------------------------------------------------------------------------------------------|------------------|
+| ConnectionString[1,2,3,4]   | string?          |
+| Namespace[1,2,3,4]          | string?          |
+| EventHubName[1,2,3,4]       | string?          |
+| ConsumerGroup[2,3,4]        | string?          |
+| BlobClientConnectionName[3] | string?          |
+| BlobContainerName[3]        | string?          |
+| PartitionId[4]              | string?          |
+| EventPosition[4]            | EventPosition    |
+| Credential[1,2,3,4]         | TokenCredential? |
+| Tracing[1,2,3,4]            | bool             |
+
+## Usage example
+
+The following example assumes that you have an Azure Event Hubs namespace and an Event Hub created and wish to configure an `EventHubProducerClient` to send events to the Event Hub. The `EventHubConsumerClient`, `EventProcessorClient`, and `PartitionReceiver`are configured in a similar manner.
+
+In the _Program.cs_ file of your project, call the `AddAzureEventHubProducerClient` extension method to register
+a `EventHubProducerClient` for use via the dependency injection container. The method takes a connection name parameter. This assumes you have included the `EntityPath` in the connection string to specify the Event Hub name.
```

Comment:
> This assumes you have included the `EntityPath` in the connection string to specify the Event Hub name.

This doesn't seem ideal, since we say we don't recommend using a connection string.

Reply:
It is also surprising since I would have expected:

```
var eventHub = builder.AddAzureEventHubs("eventhubns")
    .AddEventHub("hub");

builder.AddProject<Projects.EventHubsApi>("api")
    .WithReference(eventHub);
```

To result with the `EntityPath` in the connection string (just like it does if I'm using a database server + database, the connection string has `Database=<db_name>` in the connection string.

Is the above code supposed to use `EntityPath` in the connection string?

Reply:
@mitchdenny Brought this up already and it would mean making the "hub" it's own de-facto AspireResource to enable deep-linking like this. I replied and said the same could be said for allowing deep-linking of ASB topic and queues (a feature that isn't there either) so I think this could be punted for another release. 

As you (or maybe someone else said?) before, we do control what Resources inject as connection strings when using an AppHost, but we don't control what developers may do when managing their own connection strings. I'd rather that we "do the right thing" when a dev passes an entitypath. It seems like the path of least surprise. It's what I would expect.

Reply:
> we don't control what developers may do when managing their own connection strings. I'd rather that we "do the right thing" when a dev passes an entitypath. It seems like the path of least surprise. It's what I would expect.

I don't disagree that we should "do the right thing" when someone passes an entitypath.

The part I was commenting on is the "Usage example" the README gives assumes you are doing something we don't recommend you do - use a connection string.

Reply:
Ah, gotcha!